classes/mapworker.php

```diff
@@ -240,11 +244,46 @@ public function process_map_objects() : void {
             if (!empty($this->placestore['slicemode']) && count($notavailable) + count($impossible) > 0) {
                 $this->svgmap->add_overlay();
             }
+            // Make actual path through the map visible.
```

Comment:
puhh an five nested loop is hard to read. By now i don't see an easy idee to reduce this.

But there is one suggestion:

`


foreach ($this->placestore['paths'] as $path) {

        for ($i = 0; $i < count($order) - 1; $i++) {

            $found = false;

            for ($j = $i; $j >= 0 && !$found; $j--) {

                foreach ($cmidtoplaces[$order[$j]] as $place1) {

                    foreach ($cmidtoplaces[$order[$i + 1]] as $place2) {

                        if ($path = $this->is_path_between($place1, $place2)) {

                            $found = true;

                            $this->svgmap->set_waygone($path);

                            break 3; // break out of all loops

                        }
                    }
                }
            }
        }

`
        
I moved the outermost `foreach` loop to be the first loop and moved the `$found` variable outside the loop over `$path`, because it only needs to be checked once for each path. I added a `break 3` statement to break out of all three loops when a path is found, to avoid unnecessary iterations.
I tried this version and it seems to work.

And to be fair..  I tried https://deepai.org/chat to improve this code ;)

Reply:
That just works fine. Great idea to remove at least one loop!